docs/how_to_guide/adding_a_coalignment_method.rst

```diff
@@ -0,0 +1,44 @@
+.. _sunkit-image-how-to-guide-add-a-new-coalignment-method:
+
+****************************
+Add a New Coalignment Method
+****************************
+
+You can add a custom coalignment method in the sunkit-image package using the decorator :func:`sunkit_image.coalignment_module.register_coalignment_method`:
+
+.. code-block:: python
+
+    from sunkit_image.coalignment_module.util.decorators import register_coalignment_method
+
+    @register_coalignment_method("my_coalign")
+    def my_coalignment_method(input_array, template_array):
+        # Your coalignment code goes here
+        # This should encompass calculating the shifts, applying these shifts to the data,
+        # handling NaN values appropriately, and implementing any necessary clipping logic.
+        # Return the shifts and the coaligned array
+        return (coaligned_array, (x_shift, y_shift))
```

Comment:
Once we've firmed up what the return type of the method should be given our rethink this API, we should fix the documentation here to reflect that.